### ISSUE TITLE
Placing markers arithmetically

### ISSUE THREAD

**Issue by ChristophKuhfuss** [osmdroid/osmdroid#356]
Hello everyone,

we are currently trying to develop an indoor system for waypoints using osmdroid. We try to make the waypoints snap to multiples of 90° with respect to the building angle, but independently of which method we use to calculate the correct marker position, markers get placed very unprecisely and the result doesnt resemble straight lines at all.

For more detailed information on what we are doing and what we the problem is please check this stackoverflow question:

http://stackoverflow.com/questions/38484573/osmdroid-coordinate-snapping

In short, the markers to little leaps when dragging them and it is impossible to make them snap accurately.

**Comment by spyhunter99** [comment]
What zoom level are you operating at? There are known issues with > 20.
That is, zoom 21 and higher cause integer overflow issues.

**Comment by kuhfi** [comment]
The waypoints are intended to work for indoor navigation, so most of our work takes places in zoom level 21. The markers are still taking leaps when we reduce the zoom level to 19 however.

**Comment by MKergall** [comment]
Certainly due to the Geopoint attributes internal type defined as int (double*1E6), lacking the level of precision you need. 
See https://github.com/osmdroid/osmdroid/pull/48
Solution: help to achieve the "double" branch, to have it in V5.3.

**Comment by spyhunter99** [comment]
@MKergall great suggestion. @kuhfi if this resolve your issue, that will certainly motivate us to merge that branch in. In fact, I'll merge master into doubles now to reduce your workload. will report back when it's pushed

**Comment by spyhunter99** [comment]
pushed to the double's branch, travis is building now at https://travis-ci.org/osmdroid/osmdroid/builds/146155390 no promises that the tests will pass though

**Comment by kuhfi** [comment]
We tried building the double branch before to test it out, but we didn't manage to get it working somehow, hopefully things will work out better when the master branch is merged into the double branch.
Thanks a lot! We will test it out in the next couple days and report back if it works.

**Comment by ChristophKuhfuss** [comment]
Spent some time building from source, but finally got it working, and it seems like everything works perfectly fine. Thanks everyone!

**Comment by spyhunter99** [comment]
Wait, you got it working off of master or off of the "doubles" branch?

**Comment by ChristophKuhfuss** [comment]
Off of the "doubles" branch which we compiled from source.

2016-07-28 17:41 GMT+02:00 spyhunter99 <notifications@github.com>:
